apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -40,6 +40,13 @@ type OpenTelemetryCollectorSpec struct {
 	// MaxReplicas sets an upper bound to the autoscaling feature. If MaxReplicas is set autoscaling is enabled.
 	// +optional
 	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
+
+	// Autoscaler specifies the pod autoscaling configuration to use
+	// for the OpenTelemetryCollector workload.
+	//
+	// +optional
+	Autoscaler *AutoscalerSpec `json:"autoscaler,omitempty"`
```

Comment:
question: did we consider embedding the HPA spec in here? Or at least embed the [autoscaling behavior spec](https://pkg.go.dev/k8s.io/api/autoscaling/v2beta2#HorizontalPodAutoscalerBehavior) here?

Reply:
I was just adding enough code to be able to get an e2e test to work within the time allocated, which means we need to scale down much quicker that the default 300 seconds.

@pavolloffay what do you think?  Do I need to add the other values of PA scaling rules here?

Reply:
For my use case, I know that I would like to be able to specify policies and not just `StabilizationWindowSeconds`. Embedding only `StabilizationWindowSeconds` is going to make it so we need to add in each feature on request, making a code change for each one.